### PR TITLE
current_timestamp_p&current_timestamp

### PR DESCRIPTION
**Pull request by link-bands**
## Which issue does this PR close?

current_timestamp_p&current_timestamp

Closes #.

## Rationale for this change

Add current_timestamp_p&current_timestamp

## What changes are included in this PR?

Add current_timestamp_p&current_timestamp

## Are these changes tested?

Yes

### CODE REVIEW

File: extension/functions/src/presto.rs [hunk 1]
```diff
@@ -62,39 +69,154 @@ impl ScalarFunctionDef for CurrentTimeFunction {
...
+        let result = current_timestamp_function.execute(&[]).unwrap();
+        let result_array = result
+            .as_any()
+            .downcast_ref::<TimestampMillisecondArray>()
+            .unwrap()
+            .value(0);
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_millis() as i64;
```
Comment (wizardxz) [inline extension/functions/src/presto.rs]: Is this duplicate code with the test_current_timestamp_p()
If so, can you abstract a util function?

**Review by wizardxz — CHANGES_REQUESTED** [review]

File: extension/functions/src/presto.rs (hunk 1 above)
Comment (link-bands) [reply extension/functions/src/presto.rs]: Ok

File: extension/functions/src/presto.rs [hunk 2]
```diff
@@ -62,39 +69,184 @@ impl ScalarFunctionDef for CurrentTimeFunction {
...
+            TimeUnit::Nanosecond => SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_nanos() as i64,
+        };
+
+        assert!((result_value - now).abs() <= 1_000_000_000);
+
+        Ok(())
+    }
```
Comment (wizardxz) [inline extension/functions/src/presto.rs]:
```suggestion
    fn roughly_equal_to_now(millisecond: i64) -> bool {
        let now = SystemTime::now()
            .duration_since(UNIX_EPOCH)
            .unwrap()
            .as_millis() as i64;
        (millisecond - now).abs() <= 1
    }
```

File: extension/functions/src/presto.rs [hunk 3]
```diff
@@ -62,39 +69,184 @@ impl ScalarFunctionDef for CurrentTimeFunction {
...
+
+    enum TimeUnit {
+        Millisecond,
+        Nanosecond,
+    }
```
Comment (wizardxz) [inline extension/functions/src/presto.rs]:
```suggestion

```

File: extension/functions/src/presto.rs [hunk 4]
```diff
@@ -62,39 +69,184 @@ impl ScalarFunctionDef for CurrentTimeFunction {
...
+
+    #[tokio::test]
+    async fn test_current_timestamp() -> Result<()> {
+        test_timestamp_function(
+            |args| CurrentTimestampFunction {}.execute(args),
+            &[],
+            TimeUnit::Millisecond,
+        )
+        .await
```
Comment (wizardxz) [inline extension/functions/src/presto.rs]:
```suggestion
let current_timestamp_function = CurrentTimestampFunction {};
        let result = current_timestamp_function.execute(&[]).unwrap();
        let result_array = result
            .as_any()
            .downcast_ref::<TimestampMillisecondArray>()
            .unwrap()
            .value(0);
        assert!(roughly_equal_to_now(result_array));
```

File: extension/functions/src/presto.rs [hunk 5]
```diff
@@ -62,39 +69,184 @@ impl ScalarFunctionDef for CurrentTimeFunction {
...
+    #[tokio::test]
+    async fn test_current_timestamp_p() -> Result<()> {
+        let precision_array = Int64Array::from(vec![9]);
+        let args = vec![Arc::new(precision_array) as ArrayRef];
+        test_timestamp_function(
+            |args| CurrentTimestampPFunction {}.execute(args),
+            &args,
+            TimeUnit::Nanosecond,
+        )
+        .await
```
Comment (wizardxz) [inline extension/functions/src/presto.rs]:
```suggestion
        let function = CurrentTimestampPFunction {};
        let precision_array = Int64Array::from(vec![9]);
        let args = vec![Arc::new(precision_array) as ArrayRef];
        let result = function.execute(&args).unwrap();
        let result_array = result
            .as_any()
            .downcast_ref::<TimestampNanosecondArray>()
            .unwrap()
            .value(0);
        assert!(roughly_equal_to_now(result_array / 1_000_000));
        Ok(())
```

**Review by wizardxz — CHANGES_REQUESTED** [review]
Thanks for your update. Here are my suggestion which can simplifier your test.
In addition, you need to merge origin/release

**Review by wizardxz — APPROVED** [review]

**Pull request merged**